ddp-workspace/projects/ddp-dsm-ui/src/app/FON/components/paginator/paginator.component.ts

```diff
@@ -0,0 +1,143 @@
+  @Input() totalCount: number;
+  @Input() rowsPerPage = 10;
+  @Input() currentPageIndex: number;
+  @Input() pageSizeOptions: number[];
+  @Input() visiblePages: number;
+
+  @Output() pageChanged: EventEmitter<pageProps> = new EventEmitter();
+
+  constructor() {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.setDefaultParams();
+    this.generatePagesArray();
+  }
+
+  public get currentPageRange(): (string|number)[] {
+    const currentRange = this.filteredCurrentPageRange;
+    this.fillWithDots(currentRange);
+    return currentRange;
+  }
+
+  public changePage(page: number): void {
+    if (typeof page === 'number') {
+      this.currentPageIndex = page;
+      this.pageChanged.emit(this.pageProps);
+    }
+  }
+
+  public shiftPage(order: string): void {
+    switch (order) {
+      case SHIFT_PAGE.START:
+        this.currentPageIndex = 1;
+        break;
+      case SHIFT_PAGE.END:
+        this.currentPageIndex = this.generatePagesArray().length;
+        break;
+      case SHIFT_PAGE.ONE_MORE:
+        this.currentPageIndex = Math.min(this.currentPageIndex + 1, this.generatePagesArray().length);
+        break;
+      case SHIFT_PAGE.ONE_LESS:
+        this.currentPageIndex = Math.max(this.currentPageIndex - 1, 1);
+        break;
+      default:
+        return;
+    }
+    this.pageChanged.emit(this.pageProps);
+  }
+
+  public setRows(rows: number): void {
+    this.rowsPerPage = rows;
+    this.changePage(1);
+    this.generatePagesArray();
+  }
+
+  /* Paginator Engine */
+
+  private setDefaultParams(): void {
+    if(!this.rowsPerPage || typeof this.rowsPerPage !== 'number')
+      {this.rowsPerPage = ROWS_PER_PAGE;}
+    if(!(this.pageSizeOptions instanceof Array) || this.pageSizeOptions.length < 1)
+      {this.pageSizeOptions = PAGE_SIZE_OPTIONS;}
+    if(!this.currentPageIndex || typeof this.currentPageIndex !== 'number')
+      {this.currentPageIndex = CURRENT_PAGE_INDEX;}
+    if(!this.visiblePages || typeof this.visiblePages !== 'number')
+      {this.visiblePages = VISIBLE_PAGES;}
+  }
+
+  private get filteredCurrentPageRange(): any[] {
+    return this.generatePagesArray()
+      .filter(
+        (page, _, array) => this.displayOrNot(page, array)
+      );
+  }
+
+  private fillWithDots(pagesArray: any[]): void {
+    pagesArray
+      .forEach(
+        (page, index, array) =>
+          page - array[index + 1] < -1 &&
+          array.splice(index + 1, 0, '...')
+      );
+  }
+
+  private displayOrNot(page: number, array: number[]): boolean {
+    return (page === 1 ||
+      page === array.length ||
+      page === this.currentPageIndex ||
+      page > this.currentPageIndex - (this.visiblePages + 1) && page < this.currentPageIndex + (this.visiblePages + 1));
+  }
+
+  private get pageProps(): pageProps {
+    const from = (this.currentPageIndex - 1) * this.rowsPerPage;
+    const to = from + this.rowsPerPage;
+    return {from, to};
+  }
+
+  private generatePagesArray(): any[] {
```

Comment:
returns `number[]`

Reply:
Once I had a problem with types, that's why I needed that temporarily, looks like I forgot to change back. Thanks. fixed!